lib/rex/proto/http/client.rb

```diff
@@ -252,7 +252,11 @@ def _send_recv(req, t = -1, persist = false)
   # @return [void]
   def send_request(req, t = -1)
     connect(t)
-    conn.put(req.to_s)
```

Comment:
I originally had updated this to:

```ruby
conn.put(req.to_s, { 'Timeout' => t })
```

But it looks like from the implementation, that's a timeout per individual chunk of data sent:

Code path for the write is:

https://github.com/rapid7/rex-core/blob/1e43b174888d673a045aead891ed9c4085ad85f7/lib/rex/io/stream.rb#L191-L210

Then timed_write will go down the blocking non-timeout path by default:

https://github.com/rapid7/rex-core/blob/1e43b174888d673a045aead891ed9c4085ad85f7/lib/rex/io/stream.rb#L163-L171

---

Context for change:

Users of the http client would set a timeout of 10 seconds and expect the entire operation to timeout after 10 seconds. In reality, the current implementation is 10 second timeout for connection, no timeout for writes, and 10 seconds set for reads. This PR now limits the writes to 10 seconds as well.

It's hard to tell if the timeout missing from writes was intentional or not, original timeout added to connect here: https://github.com/rapid7/metasploit-framework/commit/5fec13a389e6a7afcbd87419300d5eceee4a98dd